src/error.rs

```diff
@@ -1,17 +1,25 @@
 use std::fmt;
 use std::error::Error;
+use std::path::PathBuf;
 
 #[derive(Debug)]
 pub enum TestError {
     ExpectedOutputDiffers,
     IoError(std::io::Error),
     ErrorParsingExitStatus(std::num::ParseIntError),
+    MissingTests(PathBuf),
+    ExpectedDirectory(PathBuf),
 }
 
-// This isn't displayed when running tests via 'cargo test' anyway
 impl fmt::Display for TestError {
-    fn fmt(&self, _: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        Ok(())
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        match self {
+            Self::ExpectedOutputDiffers => f.write_str("The expected test output differs"),
```

Comment:
Just curious - did you find a case where this was printed out or is it included for completeness?

Reply:
It's just included for completeness, I found that the errors returned by using `?` within tests is really unfriendly, so I use my own

```
Error: ExpectedOutputDiffers
thread 'goldentests' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', C:\Users\Chase\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\src\libstd\macros.rs:16:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test goldentests ... FAILED
```